src/main/java/com/example/onlineservicesemulator/Controller.java

```diff
@@ -2,44 +2,87 @@
 import javafx.scene.control.CheckBox;
 import javafx.scene.control.Label;
+import javafx.scene.control.ListView;
+import javafx.scene.input.MouseEvent;
 import javafx.scene.layout.VBox;
+import javafx.stage.Stage;
 
+import java.io.File;
+import java.io.IOException;
 import java.net.URL;
-import java.util.List;
-import java.util.ResourceBundle;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
 
 public class Controller implements Initializable {
 
+    @FXML
+    private ListView<String> filesList;
     @FXML
     private VBox checkboxList;
+    @FXML
+    private Button addFileButton;
+    @FXML
+    private Button removeFileButton;
+    private List<String> servicesNames;
+    private Map<String, List<String>> servicesAndUploadedFilesMap;
 
     @Override
     public void initialize(URL url, ResourceBundle resourceBundle) {
+        servicesNames = JSONReader.getServices();
         createAndSetCheckboxList();
+        setFilesMap();
+        addFileButton.setVisible(false);
+        setRemoveFileButtonListener();
     }
 
     public void createAndSetCheckboxList() {
-        List<String> servicesNames = JSONReader.getServices();
-        double lenghtSectionA = calculateLongestServiceString(servicesNames);
-        checkboxList.setPrefWidth(lenghtSectionA);
+        double lengthSectionA = calculateLongestServiceString(servicesNames) * 8;
+        checkboxList.setPrefWidth(lengthSectionA);
         for (String serviceName : servicesNames) {
             CheckBox checkBox = new CheckBox();
             checkBox.setId("checkbox_" + serviceName);
             Label checkBoxLabel = new Label(serviceName);
             checkBox.setGraphic(checkBoxLabel);
-            checkBoxLabel.setOnMouseClicked(event -> {
-                checkBox.setSelected(false);
-                event.consume();
-            });
+            checkBoxLabel.setOnMouseClicked(event -> onCheckboxLabelClicked(event, checkBox, checkBoxLabel, serviceName));
             checkboxList.getChildren().add(checkBox);
             checkboxList.setSpacing(5);
         }
     }
 
+    private void onCheckboxLabelClicked(MouseEvent event, CheckBox checkBox, Label checkBoxLabel, String serviceName) {
+        checkBox.setSelected(false);
+        filesList.getItems().clear();
+        String clickedService = checkBoxLabel.getText();
+        if (servicesAndUploadedFilesMap.containsKey(clickedService) && servicesAndUploadedFilesMap.get(clickedService) != null) {
+            filesList.getItems().addAll(servicesAndUploadedFilesMap.get(clickedService));
+            addFileButton.setVisible(true);
+        }
+        event.consume();
+        addFiles(serviceName);
+    }
+
+    public void setFilesMap() {
+        servicesAndUploadedFilesMap = new HashMap<>();
+        servicesNames.forEach(service -> servicesAndUploadedFilesMap.put(service, null));
+
+        List<String> arraylist1 = new ArrayList<>();
```

Comment:
why are these values hardcoded? should be the actual file names of the files uploaded

Reply:
also avoid random variable names: arraylist1, it has no meaning to what the variable is for